src/aleph/web/controllers/prices.py

```diff
@@ -164,3 +170,130 @@ async def message_price_estimate(request: web.Request):
     response = EstimatedCostsResponse.model_validate(model)
 
     return web.json_response(text=aleph_json.dumps(response).decode("utf-8"))
+
+
+async def recalculate_message_costs(request: web.Request):
+    """Force recalculation of message costs in chronological order with historical pricing.
+
+    This endpoint will:
+    1. Get all messages that need cost recalculation (if item_hash provided, just that message)
+    2. Get the pricing timeline to track price changes over time
+    3. Sort messages chronologically (oldest first)
+    4. For each message, use the pricing model that was active when the message was created
+    5. Delete existing cost entries and recalculate with historical pricing
+    6. Store the new cost calculations
+    """
+
```

Comment:
As we commented, I think that we should to add a kind of temporary token authentication based to prevent that endpoint to be used by people to DDOS the CCN as the endpoints takes time to re-process everything again.

Reply:
I have added a basic ECDSA auth system